### PR TITLE
feat: implement support for Helm v3

### LINKED ISSUE #5450: Support helmv3
**What would you like Renovate to be able to do?**
As a user of helm I would like renovate to update my Chart.yaml dependencies. They used to be in requirements.yaml, now they're located inside Chart.yaml

**Describe the solution you'd like**
Renovate should support dependencies in Chart.yaml

**Describe alternatives you've considered**
Staying on helm2, which is no longer being developed

**Additional context**
[Migrating from helm 2 to 3](https://helm.sh/docs/topics/v2_v3_migration/)

### PR DESCRIPTION
**Pull request by chgl**
This adds support for renovating Helm v3 charts. I've tested it using a personal repo, and it seems to work: https://github.com/chgl/renovate-helmv3-test/pull/1

Closes #5450

### CODE REVIEW

File: lib/manager/helmv3/extract.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        'Unsupported Chart apiVersion. Only v2 is supported.'
+      );
+      return null;
+    }
+  } catch (err) {
+    logger.debug({ fileName }, 'Failed to parse helm Chart.yaml');
+    return null;
+  }
+  let deps = [];
+  if (!(chart && is.array(chart.dependencies))) {
```
Comment (rarkins) [inline lib/manager/helmv3/extract.ts:35]:
```suggestion
  if (!is.nonEmptyArray(chart?.dependencies)) {
```

File: lib/manager/helmv3/extract.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    logger.debug({ fileName }, 'Chart has no dependencies');
+    return null;
+  }
+  deps = chart.dependencies.map((dep) => {
+    const res: PackageDependency = {
+      depName: dep.name,
+      currentValue: dep.version,
```
Comment (rarkins) [inline lib/manager/helmv3/extract.ts:42]: Should we first filter chart.dependencies for entries that have both `depName` and `currentValue`?

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/manager/helmv3/extract.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        { fileName },
+        'Unsupported Chart apiVersion. Only v2 is supported.'
+      );
+      return null;
+    }
+  } catch (err) {
+    logger.debug({ fileName }, 'Failed to parse helm Chart.yaml');
+    return null;
+  }
+  let deps = [];
```
Comment (viceice) [inline lib/manager/helmv3/extract.ts:34]:
```suggestion
  let deps: PackageDependency[] = [];
```

File: lib/manager/helmv3/extract.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+import { logger } from '../../logger';
+import { SkipReason } from '../../types';
+import { ExtractConfig, PackageDependency, PackageFile } from '../common';
+
+export function extractPackageFile(
+  content: string,
+  fileName: string,
+  config: ExtractConfig
+): PackageFile | null {
+  let chart;
```
Comment (viceice) [inline lib/manager/helmv3/extract.ts:13]: Please add a type for this

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/manager/helmv3/extract.ts (hunk 2 above)
Comment (chgl) [reply lib/manager/helmv3/extract.ts:54]: Maybe not wrong as an additional check. See https://github.com/renovatebot/renovate/pull/6820/files#diff-ad106bf0f585aa336023b11d5cc3073bR45

File: lib/manager/helmv3/extract.ts (hunk 4 above)
Comment (chgl) [reply lib/manager/helmv3/extract.ts]: Done. But I am not sure if the way I've done it is ideal. Should it be moved outside the scope of the method?

**Review by rarkins — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**